src/app/product/[categoryId]/page.tsx

```diff
@@ -0,0 +1,41 @@
+import { Suspense } from 'react';
+
+import type { Sort } from '@/app/product/_types';
+import { getProductCategories, getCategoryById } from '@/app/product/_data';
+
+import { Filters } from '@/app/product/_components/Filters';
+import { ProductListSkeleton } from '../_components/ProductListSkeleton';
+import { ProductList } from '@/app/product/_components/ProductList';
+
+interface PageParams {
+  params: { categoryId: string };
+  searchParams: {
+    page: string;
+    sort: Sort;
+    keyword: string;
+  };
+}
+
+export async function generateStaticParams() {
+  const categories = await getProductCategories();
+
+  return categories?.map((category) => ({
+    categoryId: category.id.toString(),
+  }));
+}
+
+export default async function ProductPage({ params, searchParams }: Readonly<PageParams>) {
+  const { categoryId } = params;
+
+  const categoryData = await getCategoryById(parseInt(categoryId));
+
+  return (
+    <>
+      <h1 className="text-2xl">{categoryData?.name}</h1>
+      <Filters />
+      <Suspense fallback={<ProductListSkeleton />}>
+        <ProductList category={categoryId} searchParams={searchParams} />
```

Comment:
```suggestion
        <ProductList categoryId={categoryId} searchParams={searchParams} />
```

여기도 `categoryId` 로 전달되어야 하고, type 은 `number` 로 변환된 값이여야 하겠네요.